src/actions.js

```diff
@@ -301,6 +312,23 @@ export const startWallet = (words, pin) => async (dispatch) => {
   // Set useWalletService on the redux store
   dispatch(setUseWalletService(useWalletService));
 
+  const showPinScreenForResult = async () => new Promise((resolve) => {
+    const params = {
+      cb: (_pin) => {
+        dispatch(setIsShowingPinScreen(false));
+        resolve(_pin);
+      },
+      canCancel: true,
+      screenText: t`Enter your 6-digit pin to authorize operation`,
+      biometryText: t`Authorize operation`,
+    };
+
+    NavigationService.navigate('PinScreen', params);
+
+    // We should set the global isShowingPinScreen
+    dispatch(setIsShowingPinScreen(true));
```

Comment:
Why do we need to manually hide the other component when navigating to `PinScreen`?

Shoudn't the navigation lib handle the stacking of screens on top of each other for us, which would hide the previous screen? Then just unstack it after it's done?

I'm just curious, I don't really know this lib, but that's how I thought them worked.

Reply:
I agree that the lib should handle the stacking of screens, but it is not doing that atm, maybe it is a bug on the version we are using as we are on an old version:

![image](https://user-images.githubusercontent.com/3586068/149572621-e0149242-e3fb-4a2e-89c2-e1bd7e33dcf3.png)